store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -334,46 +329,68 @@ public Set<StoreTrait> getTraits() {
-            final String optionKey = FEDERATED_STORE_PROCESSED + id;
-            final boolean isIdFound = !operation.getOption(optionKey, "").isEmpty();
-            if (!isIdFound) {
-                final HashMap<String, String> updatedOptions = isNull(operation.getOptions())
-                        ? new HashMap<>()
-                        : new HashMap<>(operation.getOptions());
-                updatedOptions.put(optionKey, getGraphId());
-                operation.setOptions(updatedOptions);
-                rtn.addAll(graphStorage.get(user, getCleanStrings(graphIdsCsv)));
-            } else {
-                final List<String> federatedStoreGraphIds = operation.getOptions()
+            String optionKey = getKeyForProcessedFedStore();
+            boolean isFedStoreIdPreexisting = addFedStoreId(operation, optionKey);
+            if (isFedStoreIdPreexisting) {
+                List<String> federatedStoreIds = operation.getOptions()
                         .entrySet()
                         .stream()
                         .filter(e -> e.getKey().startsWith(FEDERATED_STORE_PROCESSED))
                         .map(Map.Entry::getValue)
                         .collect(Collectors.toList());
-
-                final String ln = System.lineSeparator();
-                LOGGER.error("This operation has already been processed by this FederatedStore. "
-                        + "This is a symptom of an infinite loop of FederatedStores and Proxies.{}"
-                        + "This FederatedStore: {}{}"
-                        + "All FederatedStore in this loop: {}",
-                        ln, this.getGraphId(), ln, federatedStoreGraphIds.toString());
+                String ln = System.lineSeparator();
+                LOGGER.error("This operation has already been processed by this FederatedStore. " +
+                        "This is a symptom of an infinite loop of FederatedStores and Proxies.{}" +
+                        "This FederatedStore: {}{}" +
+                        "All FederatedStore in this loop: {}", ln, this.getGraphId(), ln, federatedStoreIds.toString());
+            } else if (isNull(graphIdsCsv)) {
+                LOGGER.debug("getting default graphs because requested graphIdsCsv is null");
+                rtn = getDefaultGraphs(user, operation);
+            } else {
+                String adminAuth = operation.isUserRequestingAdminUsage() ? this.getProperties().getAdminAuth() : null;
+                rtn.addAll(graphStorage.get(user, getCleanStrings(graphIdsCsv), adminAuth));
             }
+        } else {
+            LOGGER.warn("getGraphs was requested with null Operation, this will return no graphs.");
         }
+        return rtn;
+    }
+
+    private String getKeyForProcessedFedStore() {
+        return FEDERATED_STORE_PROCESSED + id;
+    }
 
+    private boolean addFedStoreId(final Operation operation, final String optionKey) {
+        boolean rtn = false;
```

Comment:
I know this was already like this, but why it is `rtn` and not something more descriptive. It looks like it could be called `hasOperationOrPayloadPreexistingFedStoreId`?

Reply:
For a "do" method which is returning a byproduct like a `isSuccess` or similar like this method. **I agree**. Its not clear what the byproduct is meant to be at a glance. so I can change this.
I think this method started life out as two methods a "do" `addID` and "get" `didIDexist` and was joined up.

for "get/create/make/fetch" methods the name of the method should be descriptive enough to know what is being returned. So having a `rtn` as a blanket standard variable name across most/all methods is clean clutter free way of writing methods. When reading methods with `rtn` it is very easy to identify the object you probably actually only care about and can easily see when it is being made/created/modified amongst a crowd of transient variables names which may be similar in noise. In the below example it should be clear what the method is getting, so having a variable called "magicSpell" might get lost amongst "stars", "magic", "phrases" and such.

``` java
public String getSuperMagicIncantationSpellPhrase(String magic, String phases, String[] stars){
    String rtn = getStartingPhrase()  //the only object you care about;
    String magic = // noise noise
    for(stars){
        if(magic){
              rtn =+ // Ah something is happening to what care about
        }
    }
    //something something
  
    return rtn; //send it!
}
```